internal/target/repository_test.go

```diff
@@ -312,3 +316,108 @@ func TestRepositoryListPermissionWhereClauses(t *testing.T) {
 		})
 	}
 }
+
+func Test_listDeletedIds(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	conn, mock := db.TestSetupWithMock(t)
+	wrapper := db.TestWrapper(t)
+	mock.ExpectQuery(
```

Comment:
I don't love having to resort to sql mocks for these tests, unfortunately I can't think of any other way to test this since the functions are unexported, and there are no concrete implementations of targets in this package.